opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -466,11 +465,38 @@ public static void addObservation(Event e, JSONObject jsonObject) {
             } catch (JSONException e1) {
                 Timber.e(e1);
             }
+        } else if (AllConstants.GPS.equals(type)) {
+            createGpsObservation(e, jsonObject, value);
         } else {
             createObservation(e, jsonObject, value);
         }
     }
 
+    private static void createGpsObservation(Event e, JSONObject jsonObject, String value) {
+        createObservation(e, jsonObject, value);
+        if (StringUtils.isNotBlank(value)) {
+            String[] valueArr = value.split(" ");
+            String formSubmissionFieldPrefix = getString(jsonObject, KEY);
+            if (valueArr.length >= 2) {
+                String latitude = valueArr[0];
+                String longitude = valueArr[1];
+                addObs(e, formSubmissionFieldPrefix + "_" + AllConstants.GpsConstants.LATITUDE, AllConstants.TEXT, Collections.singletonList(latitude));
```

Comment:
These operations : `formSubmissionFieldPrefix + "_" +`, `AllConstants.TEXT` and `Collections.singletonList()` are still repeated. I think it's possible to simplify this by passing an event and `formSubmissionFieldPrefix` and suffix to a method that will perform the duplicated functionality.

Reply:
And since the method call would be relatively the same (minus the suffix), an array can be used to store the suffixes and loop through them while making the call.